### PR TITLE
Extend iotrace test API

### PR DESCRIPTION
**Pull request by Ostrokrzew**
Extend ***iotrace*** module
Modify `iotrace` module to be more useful.
Change some methods to be static.
Update usage of changed methods.
Add new methods.
Add parameters to existing methods.
Refactor code

### CODE REVIEW

**Comment by tomaszrybicki** [comment]
retest this please

**Comment by tomaszrybicki** [comment]
retest this please

File: tests/utils/iotrace.py [hunk 1]
```diff
@@ -120,132 +184,282 @@ def get_sort_key(element):
...
+        subrange_range = range(1, 9223372036854775808)
+        if subrange_start and subrange_end:
+            if subrange_start > subrange_end:
+                subrange_start, subrange_end = subrange_end, subrange_start
+
+        command = 'iotrace' + (' -P' if shortcut else ' --trace-parser')
+        command += ' -B' if shortcut else ' --lba-histogram'
+        command += (' -p ' if shortcut else ' --path ') + f'{trace_path}'
+        if bucket_size is not None \
+                and int(bucket_size.get_value(Unit.Byte)) in bucket_size_range:
```
Comment (imjfckm) [inline tests/utils/iotrace.py]: I don't think it's the best idea to just silently discard arguments that are not in range. What I'd expect is that if I don't specify the argument I get the default, but if I specify wrong argument then raise exception.

File: tests/utils/iotrace.py [hunk 2]
```diff
@@ -120,132 +184,282 @@ def get_sort_key(element):
...
+            raise Exception("Invalid traces")
+
+        return parse_json(output.stdout)
+
+    @staticmethod
+    def remove_traces(prefix: str,
+                      force: bool = False,
+                      shortcut: bool = False):
+        """
+        Returns trace path of the most recent trace
```
Comment (imjfckm) [inline tests/utils/iotrace.py]: That's not what this function does

File: tests/utils/iotrace.py [hunk 3]
```diff
@@ -8,105 +8,169 @@
...
                 bdevs.append(disk.system_path)
 
-        opts = ""
-        if buf:
-            buf = int(buf.get_value(Unit.MiB))
-            opts += f" -b {buf}"
-
-        TestRun.executor.run_in_background('iotrace -S -d ' + ','.join(bdevs) + opts)
+        buffer_range = range(1, 1025)
+        trace_file_size_range = range(1, 100000001)
```
Comment (mmkayPL) [inline tests/utils/iotrace.py]: Similar issue of silent parameter discard happens here

File: tests/utils/iotrace.py [hunk 4]
```diff
@@ -120,132 +188,288 @@ def get_sort_key(element):
...
+        :return: io-tracer version
+        :rtype: list of dictionaries
+        :raises Exception: if command fails
+        """
+        output = parse_json(TestRun.executor.run_expect_success(
+            'iotrace' + (' -V ' if shortcut else ' --version ')))
+        version = (output[0]["system"] + " version: " + output[0]["trace"]
+                   + "\n" + output[1]["system"] + " version: "
+                   + output[1]["trace"] + "\n")
+        return version
```
Comment (tomaszrybicki) [inline tests/utils/iotrace.py]: Doesn't this return a string?
Also I'd drop the custom display format, and either return 'output'
or return the string with 'output[1]["trace"]' - to make parsing this return more intuitive

Comment (Ostrokrzew) [reply tests/utils/iotrace.py]: I decided to return a dictionary, it would be more readable and also easy to parse.

**Comment by imjfckm** [comment]
Maybe as a general remark - I'd consider adding some spacing between code blocks/test steps etc. This wall of code is sometimes difficutl to process

File: tests/utils/iotrace.py [hunk 5]
```diff
@@ -8,244 +8,461 @@
...
+            if subrange_start > subrange_end:
+                subrange_start, subrange_end = subrange_end, subrange_start
+
+        command = 'iotrace' + (' -P' if shortcut else ' --trace-parser')
+        command += ' -B' if shortcut else ' --lba-histogram'
+        command += (' -p ' if shortcut else ' --path ') + f'{trace_path}'
+        if bucket_size is not None:
+            if int(bucket_size.get_value(Unit.Byte)) not in bucket_size_range:
+                raise CmdException(f"Given size is out of range {bucket_size_range}.")
+            command += ' - b ' if shortcut else ' --bucket-size '
```
Comment (mmichal10) [inline tests/utils/iotrace.py]: I believe the shouldn't be space in `- b`

File: tests/utils/iotrace.py [hunk 6]
```diff
@@ -8,244 +8,471 @@
...
+        return TestRun.executor.run_expect_success(
+            'iotrace' + (' -H ' if shortcut else ' --help ')).stdout
+
+    @staticmethod
+    def version(shortcut: bool = False) -> dict:
+        """
+        :param shortcut: Use shorter command
+        :type shortcut: bool
+        :return: io-tracer version
+        :rtype: dictionary
```
Comment (tomaszrybicki) [inline tests/utils/iotrace.py]: Please also add expected key values to doc - because this is not clear, even looking at implementation

**Pull request merged**